app/lib/kafka.server.ts

```diff
@@ -13,7 +13,7 @@ import { domain, getEnvOrDieInProduction } from './env.server'
 
 const client_id = getEnvOrDieInProduction('KAFKA_CLIENT_ID') ?? ''
 const client_secret = getEnvOrDieInProduction('KAFKA_CLIENT_SECRET')
-const kafka = new Kafka({
+const kafka: Kafka | undefined = new Kafka({
```

Comment:
No type definition needed here because you are assigning a value to it and it can infer the type. Also, it can't be undefined here.

```suggestion
const kafka = new Kafka({
```